src/olakaisdk/context.py

```diff
@@ -0,0 +1,143 @@
+"""Context manager for metadata injection into LLM telemetry."""
+
+import contextvars
+from typing import Dict, Optional
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+
+
+# Thread-safe context storage using contextvars
```

Comment:
This solution is great, because on top of being thread-safe, it is also coroutine-safe (asyncio guarantees that the context is properly passed into child coroutines and tasks, whatever the constructor used to build and schedule them)